### PR TITLE
Add support to use corpus as dataset for vector search

### PR DESCRIPTION
**Pull request by VijayanB**
### Description
Accept corpus as input for bulk, query and neighbor dataset.
Here, data_set_format will be used for corpus format as well.

### Issues Resolved

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by VijayanB** [comment]
This depends on #202

File: vectorsearch/params/nmslib-sift-128-l2.json [hunk 1]
```diff
@@ -15,7 +15,12 @@
     "target_index_force_merge_timeout": 45.0,
     "hnsw_ef_search": 100,
     "hnsw_ef_construction": 100,
+
```
Comment (IanHoang) [inline vectorsearch/params/nmslib-sift-128-l2.json:18]: Was this new line intended?

File: vectorsearch/params/lucene-sift-128-l2.json [hunk 2]
```diff
@@ -15,7 +15,12 @@
     "target_index_force_merge_timeout": 45.0,
     "hnsw_ef_search": 100,
     "hnsw_ef_construction": 100,
+
```
Comment (IanHoang) [inline vectorsearch/params/lucene-sift-128-l2.json:18]: Is this new line intended?

File: vectorsearch/params/faiss-sift-128-l2.json [hunk 3]
```diff
@@ -15,7 +15,12 @@
     "target_index_force_merge_timeout": 45.0,
     "hnsw_ef_search": 100,
     "hnsw_ef_construction": 100,
+
```
Comment (IanHoang) [inline vectorsearch/params/faiss-sift-128-l2.json:18]: Same as below comments

**Review by IanHoang — APPROVED** [review]
Overall LGTM

File: vectorsearch/params/lucene-sift-128-l2.json (hunk 2 above)
Comment (IanHoang) [reply vectorsearch/params/lucene-sift-128-l2.json:18]: NVrm I see the original file

**Pull request merged**